src/testdir/test_json.vim

```diff
@@ -242,7 +242,7 @@ func Test_js_decode()
   call assert_equal(s:varnr, js_decode(s:jsonnr))
   if has('float')
     call assert_equal(s:varfl, js_decode(s:jsonfl))
-    call assert_equal(s:varinf, js_decode(s:jsoninf))
+    call assert_true(isinf(js_decode(s:jsoninf)))
```

Comment:
I just noticed this:
```
:echo js_decode('Infinity')
inf
```
```
:echo js_decode('-Infinity')
E474: Invalid argument
E474: Invalid argument
E474: Invalid argument
v:none
```
Treating js_decode('-Infinity') as an error looks like a bug to me, but it's unrelated to your change.